src/components/interactive-builder/modals/question/question-form/question/question.component.tsx

```diff
@@ -59,11 +60,14 @@ const Question: React.FC<QuestionProps> = ({ checkIfQuestionIdExists }) => {
         )}
         required
       />
+
       <Select
         value={formField?.type ?? 'control'}
-        onChange={(event: React.ChangeEvent<HTMLSelectElement>) =>
-          setFormField({ ...formField, type: event.target.value })
-        }
+        onChange={(event) => {
+          const newType = event.target.value;
+          const cleaned = cleanFormFieldForType({ ...formField, type: newType }, newType);
+          setFormField(cleaned);
+        }}
```

Comment:
Its better to not do this because this would create a new function for every render (see our docs on guidance for performance - https://o3-docs.openmrs.org/docs/coding-conventions/performance). It would be better if we do 
```
const handleQuestionTypeChange = useCallback(() => 
 {
    // whatever the onchange handles
  }, []);
```
and then,
```suggestion
        onChange={handleQuestionTypeChange}
```

Reply:
Ok understood..!